File: scratch/patch_sumber_metric.py
from pathlib import Path

p = Path(r"c:\laragon\www\coverage-area\resources\js\Pages\Monitoring\Tabs\DashboardTab.jsx")
text = p.read_text(encoding="utf-8")
old = """                                                                                        {metricCells(
                                                                                            item,
                                                                                        )}"""
new = """                                                                                        {metricCells(
                                                                                            item,
                                                                                            {
                                                                                                componentType:
                                                                                                    "sumber_dana",
                                                                                                componentLabelPrefix:
                                                                                                    "Sumber Dana",
                                                                                            },
                                                                                        )}"""
count = text.count(old)
print("matches", count)
if count >= 1:
    # Prefer occurrence after Sumber Dana label
    marker = "Sumber\n                                                                                    Dana"
    idx = text.find(marker)
    if idx < 0:
        marker = "Sumber Dana"
        # find second metricCells bare after first jenjang already patched
        idx = 0
    pos = text.find(old, max(0, idx))
    print("pos", pos, "idx", idx)
    if pos >= 0:
        text = text[:pos] + new + text[pos + len(old):]
        p.write_text(text, encoding="utf-8")
        print("ok")
    else:
        print("pos not found")
else:
    print("no bare metricCells")

File: scratch/patch_sumber_metric.php
<?php
$path = __DIR__ . '/../resources/js/Pages/Monitoring/Tabs/DashboardTab.jsx';
$text = file_get_contents($path);
$old = "                                                                                        {metricCells(\n                                                                                            item,\n                                                                                        )}";
$new = "                                                                                        {metricCells(\n                                                                                            item,\n                                                                                            {\n                                                                                                componentType:\n                                                                                                    \"sumber_dana\",\n                                                                                                componentLabelPrefix:\n                                                                                                    \"Sumber Dana\",\n                                                                                            },\n                                                                                        )}";
$count = substr_count($text, $old);
echo "matches=$count\n";
if ($count < 1) {
    exit(1);
}
// replace only first remaining bare call (sumber)
$pos = strpos($text, $old);
$text = substr_replace($text, $new, $pos, strlen($old));
file_put_contents($path, $text);
echo "ok at $pos\n";
